src/core/server/metrics/integration_tests/metrics_service.test.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * and the Server Side Public License, v 1; you may not use this file except in
+ * compliance with, at your election, the Elastic License or the Server Side
+ * Public License, v 1.
+ */
+
+import * as kbnTestServer from '../../../test_helpers/kbn_server';
+import { InternalCoreSetup } from '../../internal_types';
+import { Root } from '../../root';
+import { OpsMetrics } from '../types';
+
+const otherTestSettings = {
+  ops: {
+    interval: 500,
+  },
+  logging: {
+    silent: true, // set "true" in kbnTestServer
+    appenders: {
+      'custom-console': {
+        kind: 'console',
+        layout: {
+          highlight: false,
+          kind: 'pattern',
+          pattern: '%message|%meta',
+        },
+      },
+    },
+    root: {
+      appenders: ['custom-console', 'default'],
+      level: 'warn',
+    },
+    loggers: [
+      {
+        context: 'metrics.ops',
+        appenders: ['custom-console'],
+        level: 'debug',
+      },
+    ],
+  },
+  plugins: {
+    initialize: false,
+  },
+};
+
+describe('metrics service', () => {
+  let root: Root;
+  let coreSetup: InternalCoreSetup;
+  let mockConsoleLog: jest.SpyInstance;
+  let testData: OpsMetrics;
+
+  beforeAll(async () => {
+    mockConsoleLog = jest.spyOn(global.console, 'log');
+    mockConsoleLog.mockClear();
+  });
+
+  afterEach(() => {
+    mockConsoleLog.mockClear();
+  });
+
+  afterAll(async () => {
+    mockConsoleLog.mockRestore();
+    if (root) {
+      await root.shutdown();
+    }
+  });
+
+  describe('setup', () => {
+    it('returns ops interval and getOpsMetrics$ observable', async () => {
+      root = kbnTestServer.createRoot({ ...otherTestSettings });
+      coreSetup = await root.setup();
+      expect(coreSetup.metrics).toHaveProperty(
+        'collectionInterval',
+        otherTestSettings.ops.interval
+      );
+      expect(coreSetup.metrics).toHaveProperty('getOpsMetrics$');
+      await root.shutdown();
+    });
+  });
+
+  describe('ops metrics logging configuration', () => {
+    it('does not log with logging set to quiet', async () => {
+      root = kbnTestServer.createRoot({ logging: { quiet: true } });
+      coreSetup = await root.setup();
+
+      coreSetup.metrics.getOpsMetrics$().subscribe((opsMetrics) => {
+        testData = opsMetrics;
+      });
+
```

Comment:
nit - You can probably remove this from most of these tests, since there are only a few places where you are actually performing assertions against `testData`
```suggestion
```

Reply:
Done